src/_P016_IR.ino

```diff
@@ -331,10 +328,10 @@ boolean Plugin_016(byte function, struct EventStruct *event, String& string)
   // If we got a human-readable description of the message, display it.
           if (description != "") addLog(LOG_LEVEL_INFO, description);
 #endif  // Extended Messages
-          // Output RAW timing info of the result.
-          //log += resultToTimingInfo(&results);  //not showing up nicely in the web log... Maybe send them to serial?
-          // Output the results as source code
-          //  log += resultToSourceCode(&results); //not showing up nicely in the web log... Maybe send them to serial?
+          
+          unsigned long IRcode = results.value;
+          UserVar[event->BaseVarIndex] = (IRcode & 0xFFFF);
+          UserVar[event->BaseVarIndex + 1] = ((IRcode >> 16) & 0xFFFF);
```

Comment:
It seems the result is always set, regardless if it was useful? (it has passed the "decode" step)
I noticed there is no `READ` function here in the plugin, only a 10/sec function.
So there is only an event, but it cannot be sent to any controller? (the read does fetch a last reading and hand it over to a controller)

Reply:
> It seems the result is always set, regardless if it was useful?

Yes because it does not matter if the library can't decode it. I would prefer a result that could go back at the condroller of the type "NEC,ASDF12"

> So there is only an event, but it cannot be sent to any controller? 
I think I do not know what you mean. I just know that this value can be used in rules.

Reply:
What I meant was that it can only be used in rules.
Other plugins will also be able to signal when a new value was read and that will then be sent to a controller.
But I guess the IR is so ad-hoc, it should react immediately and thus is the normal read function a bit useless.
I was just wondering while reading the code. Nothing you should change, but maybe you had some ideas about it yourself :)

Reply:
Can we anyway send the decoded value at the condroller? I remember that we could not send Strings back, is that correct?

Reply:
> 
> 
> Ah that's probably the main reason. Controllers only accept floats/ints

How easy/hard would it be to implement this? I belive that it has value, at least for this plug in it has.

Reply:
That's quite a big change in how ESPeasy is handling data.
I guess all parts that handle data should be changed to accept strings.

Also it may take quite a bit more RAM